### PR TITLE
add flake.nix including github action to update it

### LINKED ISSUE #12: Suggestion: provide a flake
I had the following idea:

This repo could provide a flake similar to this:

```
{
 inputs = {
  database = {
   type = "tarball";
   url = "https://github.com/Mic92/nix-index-database/releases/download/2022-05-22/index-x86_64-linux";   
   };
 }
 outputs = {database} : {
  inherit database;
 }
}
```

The github action could update the url and the lockfile.
This way flake users could include the database purely in their system and update it with `flake update`.

What do you think?

### PR DESCRIPTION
**Pull request by maralorn**
This should fix #12.

Is there an easy way for me to test the github action?

cc @lilyball

### CODE REVIEW

**Comment by Mic92** [comment]
Yes, this action can be started manually: If you want to test this go to https://github.com/maralorn/nix-index-database/actions/workflows/update.yml
and click on the button next to `This workflow has a workflow_dispatch event trigger.`

File: .github/workflows/update.yml [hunk 1]
```diff
@@ -58,3 +58,43 @@ jobs:
...
+              sha256 = "${{ steps.prefetch-linux.outputs.hash }}";
+            };
+            packages.x86_64-darwin.database = nixpkgs.legacyPackages.x86_64-darwin.fetchurl {
+              url = "https://github.com/Mic92/nix-index-database/releases/download/${{ steps.date.outputs.date }}/index-x86_64-darwin";
+              sha256 = "${{ steps.prefetch-darwin.outputs.hash }}";
+            };
+          };
+        }
+        EOF
+    - name: commit and push flake.nix
```
Comment (Mic92) [inline .github/workflows/update.yml:94]: I would use a github action here that provides this functionality: 

```
      - name: Create Pull Request
        id: cpr
        uses: peter-evans/create-pull-request@v4
        with:
          commit-message: Update flakes
          branch: flake-updates
          delete-branch: true
          title: 'Upgrade flakes'
```

File: .github/workflows/update.yml [hunk 2]
```diff
@@ -58,3 +58,43 @@ jobs:
...
+    - name: prefetch linux index
+      id: prefetch-linux
+      run: echo "::set-output name=hash::$(nix shell nixpkgs#nix-prefetch -c nix-prefetch-url https://github.com/Mic92/nix-index-database/releases/download/${{ steps.date.outputs.date }}/index-x86_64-linux)"
+    - name: prefetch darwin index
+      id: prefetch-darwin
+      run: echo "::set-output name=hash::$(nix shell nixpkgs#nix-prefetch -c nix-prefetch-url https://github.com/Mic92/nix-index-database/releases/download/${{ steps.date.outputs.date }}/index-x86_64-darwin)"
+    - name: overwrite flake.nix
+      run: |
+        cat > flake.nix << EOF
+        {
```
Comment (Mic92) [inline .github/workflows/update.yml:80]:
```suggestion
        # this file is autogenerated by .github/workflows/update.yml
        {
```

**Comment by maralorn** [comment]
This should now be working.

File: flake.nix [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# this file is autogenerated by .github/workflows/update.yml
+{
+  description = "nix-index database";
+  inputs.nixpkgs.url = "nixpkgs/nixos-22.05"; # pinned to stable until https://github.com/NixOS/nixpkgs/issues/177660 is fixed.
```
[the remark this answers is not in the document]
Comment (maralorn) [reply flake.nix:4]: Yeah, the pinning issue is unfortunate. I found out though what the problem is: https://github.com/NixOS/nixpkgs/issues/177660

pennae suggested either zipping the database in postFetch or downloading a compressed file to workaround Nix finding hashes in there.

The problem with builtins.fetchurl is that it does not take a hash as input and is therefor impure. I think it’s even disallowed in flakes.

[the remark this answers is not in the document]
Comment (lilyball) [reply flake.nix]: The docs are wrong, it actually can take an attrset like `fetchTarball` does. The error you get in pure evaluation mode even hints at this:

```console
❯ nix eval --expr 'builtins.fetchurl "http://example.com"'
error: in pure evaluation mode, 'fetchurl' requires a 'sha256' argument
```

[the remark this answers is not in the document]
Comment (maralorn) [reply flake.nix]: thank you. I now have a much better solution

**Comment by AnusraMana** [comment]
@

**Pull request merged**